tests/src/Context/ContentTypeContext.php

```diff
@@ -0,0 +1,203 @@
+    $field_info = array();
+    $field_instance_info = array();
+
+    $field_instance_info['entity_type'] = 'node';
+    $field_instance_info['bundle'] = $arg1;
+
+    // Assign fields to user before creation.
+    foreach ($settings->getRowsHash() as $setting_name => $value) {
+      switch ($setting_name) {
+        case 'Label':
+          $field_instance_info['label'] = $value;
+          break;
+
+        case 'Name':
+          $field_name = 'field_' . $value;
+          $is_field_exiting = field_info_field($field_name);
+          if ($is_field_exiting) {
+            throw new \InvalidArgumentException(sprintf('The field "%s" already exists and cannot be used for test purpose.', $setting_name));
+          }
+          $field_info['field_name'] = $field_name;
+          $field_instance_info['field_name'] = $field_name;
+          break;
+
+        case 'Type':
+          $field_info['type'] = $value;
+          break;
+
+        case 'Cardinality':
+          $field_info['cardinality'] = ($value == "unlimited") ? -1 : $value;
+          break;
+
+        case 'Widget':
+          $field_instance_info['widget']['type'] = $value;
+          break;
+
+        case 'Allowed values':
+          $values = explode('::', $value);
+          foreach ($values as $allowed_value) {
+            $allowed_option = $allowed_value;
+            $allowed_label = $allowed_value;
+            $parsed_value = explode('>', $allowed_value);
+            if (count($parsed_value) == 2) {
+              $allowed_option = $parsed_value[0];
+              $allowed_label = $parsed_value[1];
+            }
+            $allowed_values[$allowed_option] = $allowed_label;
+          }
+          $field_info['settings']['allowed_values'] = $allowed_values;
+          break;
+
+        case 'Translatable':
+          $field_info['translatable'] = (bool) $value;
+          break;
+
+        case 'Default values':
+          break;
+
+        case 'description':
+          $field_instance_info['label'] = $value;
+          break;
+
+        case 'Required':
+          $field_instance_info['required'] = (bool) $value;
+          break;
+      }
+    }
+    field_create_field($field_info);
+    field_create_instance($field_instance_info);
+  }
+
+  /**
+   * Add a field group in the content type view and assign fields as children.
+   *
+   * @param string $arg1
+   *    The machine name of the content type to which attaching the field.
+   * @param TableNode $settings
+   *    Two columns table containing field settings.
+   *
+   * @Given a field group with the following settings is added to the :arg1 type view:
+   */
+  public function aFieldGroupWithTheFollowingSettingsIsAddedToTheTypeView($arg1, TableNode $settings) {
+    $group_machine_type = 'fieldset';
+    $mode = 'default';
+    $group_children = array();
+    // TODO : implement the fully support of the group type, missing
```

Comment:
I will not because that it is easier to work on this method while working on concrete cases, and adding a new element in it when it is necessary.

The todo comment is there for just reminding the method is fully implemented and we are aware  of it.